src/indicators/drought/biomasse.py

```diff
@@ -0,0 +1,335 @@
+    )
+
+    # calculate biomasse (cumulative sum of DMP across the season
+    # starting with the start_dekad and ending at start_dekad - 1
+    df_merged.sort_values(
+        by=admin_cols + ["year", "dekad"], inplace=True, ignore_index=True
+    )
+    df_merged["season"] = df_merged["year"] + (
+        df_merged["dekad"] >= start_dekad
+    )
+    df_merged["biomasse"] = df_merged.groupby(by=admin_cols + ["season"])[
+        "DMP"
+    ].apply(lambda x: x.cumsum() * 365.25 / 36)
+    df_merged["biomasse_anomaly"] = (
+        100 * df_merged["biomasse"] / df_merged["biomasse_mean"]
+    )
+
+    # re-arrange columns
+    df_merged = df_merged[
+        admin_cols
+        + [
+            "year",
+            "dekad",
+            "DMP_MEA",
+            "DMP",
+            "biomasse_mean",
+            "biomasse",
+            "biomasse_anomaly",
+        ]
+    ]
+
+    # save file to processed filepath
+    processed_path = _get_processed_path(
+        admin=admin, iso3=None, start_dekad=start_dekad
+    )
+    df_merged.to_csv(processed_path, index=False)
+
+    return df_merged
+
+
+def load_biomasse_data(
+    admin: AdminArgument = "ADM2",
+    iso3: Union[bool, str] = None,
+    start_dekad: int = 10,
+    reprocess: bool = False,
+    redownload: bool = False,
+):
+    """Load biomasse processed data
+
+    Load the data that has been processed at the
+    specified ADM2 level and starting dekad.
+
+    Args:
+        admin: Admin level, one of 'ADM0', 'ADM1',
+            or 'ADM2'.
+        start_dekad: Starting dekad for annual
+            calculations.
+        reprocess: Boolean, whether to reprocess.
+        redownload: Boolean, whether to reprocess.
+        file_descriptor: If not None, file descriptor
+            to access non-admin file.
+
+    Returns:
+        Pandas data frame.
+    """
+    if reprocess:
+        return process_dmp(
+            admin=admin, start_dekad=start_dekad, redownload=redownload
+        )
+    else:
+        _check_admin(admin)
+        processed_path = _get_processed_path(
+            admin=admin, iso3=iso3, start_dekad=start_dekad
+        )
+        return pd.read_csv(processed_path)
+
+
+def aggregate_biomasse(
+    admin_pcodes: List[str],
+    admin: AdminArgument = "ADM2",
+    start_dekad: int = 10,
+    iso3: Union[bool, str] = None,
+):
+    """Aggregate biomasse data to set of areas
+
+    While the generic processed administrative
+    biomasse data is useful, sometimes we want
+    to aggregate the biomasse values to a set
+    of administrative areas. This should be done
+    by summing up the mean biomasse for each dekad
+    for those areas and the reported biomasse for
+    each year and dekad, then recalculating the
+    anomaly.
+    """
+    _check_admin(admin)
+    df = load_biomasse_data(admin=admin, start_dekad=start_dekad)
+    admin_col = admin.replace("ADM", "admin") + "Pcod"
+    df_subset = df[df[admin_col].isin(admin_pcodes)]
```

Comment:
maybe add a comment here on descripttion that the exact boundaries of the admin shapes that Biomasse uses are not known? So solely concat based on pcode?

Reply:
In practice shouldn't give a large difference but think good to note somewhere as a caveat

Reply:
In this case, we are concating based on Pcode because it's the easiest way to join, but they are the same geographic areas as area of interest. There are some areas outside our area of interest that don't match at the ADMIN2 level but the ADMIN2's completely align for our area of interest in this case.